vars/multicloudDeploy.groovy

```diff
@@ -81,9 +91,27 @@ void call(parameters = [:]) {
                         mtaPath: script.commonPipelineEnvironment.mtarFilePath,
                         deployTool: deployTool
                     )
+                    if (runInIsolatedWorkspace) {
+                        deploymentUtils.stashStageFiles(script, stageName)
+                    }
+                }
+                if (runInIsolatedWorkspace){
+                    deployments["Deployment ${index}"] = {
+                        if (env.POD_NAME) {
+                            dockerExecuteOnKubernetes(script: script, containerMap: ContainerMap.instance.getMap().get(stageName) ?: [:]) {
+                                deployment.call()
+                            }
+                        } else {
+                            node(env.NODE_NAME) {
+                                deployment.call()
+                            }
+                        }
+                    }
+                    index++
```

Comment:
this is the same in the if and the else block, right? could ++ the index outside of the if?